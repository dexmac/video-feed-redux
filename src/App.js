import React, { Component } from "react";
import { NAVBAR_BRAND } from "./constants/constants";
import VideoFeedManager from "./containers/VideoFeedManager";
import NavigationBar from "./components/NavigationBar";
import { Provider } from "react-redux";
import "./styles/App.css";
import PropTypes from "prop-types";

export class App extends Component {
  render() {
    return (
      <Provider store={this.props.store}>
        <div className="App">
          <NavigationBar brand={NAVBAR_BRAND} />
          <VideoFeedManager />
        </div>
      </Provider>
    );
  }
}

App.propTypes = {
  store: PropTypes.object
};

export default App;
