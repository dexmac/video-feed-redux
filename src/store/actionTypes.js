export const FETCHING_FEED_ITEMS = "feed.FETCHING_FEED_ITEMS";
export const FEED_ITEMS_FETCHED = "feed.FEED_ITEMS_FETCHED";
export const FEED_ITEMS_FETCH_ERROR = "feed.FEED_ITEMS_FETCH_ERROR";

export const LOADING_VIDEO = "item.LOADING_VIDEO";
export const VIDEO_LOADING_SUCCESS = "item.VIDEO_LOADING_SUCCESS";
export const VIDEO_LOADING_ERROR = "item.VIDEO_LOADING_ERROR";

export const UPDATE_SOURCE_SELECTION = "sourceSelect.VIDEO_SOURCE_SELECTION";
