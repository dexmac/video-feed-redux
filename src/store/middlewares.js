import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const middlewares = [];
const logger = createLogger({ collapsed: true });
middlewares.push(logger);
middlewares.push(thunkMiddleware);

export default middlewares;
