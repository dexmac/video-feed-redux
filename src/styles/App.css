.App {
}

.App-header {
  background-color: gray;
}

.navbar-brand {
  -webkit-user-select: none; /* Safari */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* IE10+/Edge */
  user-select: none; /* Standard */
}

/* Navbar CSS */
.navbar-header {
  float: left;
  padding: 15px;
  text-align: center;
  width: 100%;
}
.navbar-default .navbar-brand {
  color: gray;
  float: none;
}

/* Video feed item CSS */
.player-wrapper {
  text-align: center;
  position: relative;
  padding-top: 56.25%; /* 720 / 1280 = 0.5625 */
}

.react-player {
  position: absolute;
  top: 0;
  left: 0;
}

.video-item-header {
  font-weight: bold;
  padding: 30px 5px 30px 5px;
  width: 100%;
}

.video-item-title {
  float: left;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.video-item-views-number {
  float: right;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

/* Video sources filter select options CSS */
.select-videos {
  width: 100%;
}

.select-videos-caption{
  text-align: center;
  padding-bottom: 10px;
}

.select-videos-options {
  text-align-last:center;
  width: 100%;
  border: 5px solid #ccc;
  border-radius: 3px;
  overflow: hidden;
  background: #fafafa;
  margin-bottom: 12px;
}

.select-videos-options select {
  padding: 25px 8px;
}

.select-videos-options select:focus {
  outline: none;
}

/* Error Message Centering CSS */

.video-load-error-text {
  text-align: center;
  position:relative;
  border: 1px solid red;
  top:0;
  right:0;
  bottom:0;
  left:0;
  margin:auto;
  background: white;
  color: black;
  font-weight: bold;
  font-size: 1.5em;
  word-break: break-all;
}

/* CSS for video item title on iPhone 5 */
@media only screen
and (min-device-width : 320px)
and (max-device-width : 360px) {
  .video-item-title {
    max-width: 200px;
  }
  .video-item-views-number {
    max-width: 110px;
  }
}

/* CSS for video item title for anything smaller than iPhone 5 */
@media only screen and (max-width: 300px)  {
  .video-item-views-number {
    display: none;
  }
}

/** Loader Spinner CSS
* From https://projects.lukehaas.me/css-loaders/
*/

.loader,
.loader:after {
  border-radius: 50%;
  width: 7.5em;
  height: 7.5em;
}
.loader {
  margin: 60px auto;
  font-size: 10px;
  position: relative;
  text-indent: -9999em;
  border-top: 1.1em solid rgba(24,255,247, 0.2);
  border-right: 1.1em solid rgba(24,255,247, 0.2);
  border-bottom: 1.1em solid rgba(24,255,247, 0.2);
  border-left: 1.1em solid #18fff7;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: load8 1.1s infinite linear;
  animation: load8 1.1s infinite linear;
}
@-webkit-keyframes load8 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
@keyframes load8 {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
