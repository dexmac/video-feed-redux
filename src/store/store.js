import { createStore, combineReducers, applyMiddleware } from "redux";
import middlewares from "./middlewares";
import feed from "./feed/reducer";
import item from "./item/reducer";
import sourceSelect from "./sourceSelect/reducer";

const reducer = combineReducers({
  feed,
  item,
  sourceSelect
});

/**
 * Main function to create the store
 */
const store = createStore(reducer, applyMiddleware(...middlewares));

export default store;
