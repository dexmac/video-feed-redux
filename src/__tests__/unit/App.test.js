import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import store from "../../store/store";
import { App } from "../../App";

it("renders correctly && matches snapshot", () => {
  const tree = shallow(<App store={store} />);
  expect(tree).toMatchSnapshot();
});
