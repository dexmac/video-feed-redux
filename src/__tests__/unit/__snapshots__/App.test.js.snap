// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly && matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="App"
  >
    <NavigationBar
      brand="Video Feed"
    />
    <Connect(VideoFeedManager) />
  </div>
</Provider>
`;
